Remove MobX option doesn't remove all mobx code

### Describe the bug

# Bug: Ignite "Remove MobX" Does Not Fully Remove MobX Code

## Description

When using the Ignite CLI tool and selecting the option to remove MobX (MobX-State-Tree), the MobX code is not fully removed from the project. This issue is observed in files such as the AppNavigator boilerplate and other areas of the codebase.

## Steps to Reproduce

1. Generate a new project or use an existing project with Ignite that includes MobX/MST macros.
2. Run the Ignite tool and select the option to remove MobX.
3. Observe that MobX-related code (e.g., in `AppNavigator.tsx`) is not completely removed.

## Additional Context

- Attempting to use multiple macros on a single line, such as:
  ```js
  // @demo remove-block-start @mst remove-block-start
  ```
  does not seem to work as expected. This may be related to the issue, as only one macro is processed per line.

My attempt: https://github.com/objectiveSee/ignite/tree/feat(boilerplate)-improve-mobx-remove

## Example Errors After Removal

After running the Ignite MobX removal tool, the following TypeScript/ESLint errors were encountered, indicating places where MobX code or related imports were not fully removed:

```
/Users/username/workspace/project-folder/project-name/app/models/AuthenticationStore.ts
  1:46  error  Unable to resolve path to module 'mobx-state-tree'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/models/Episode.ts
  1:58  error  Unable to resolve path to module 'mobx-state-tree'              import/no-unresolved
  2:35  error  Unable to resolve path to module './helpers/withSetPropAction'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/models/EpisodeStore.ts
  1:46  error  Unable to resolve path to module 'mobx-state-tree'              import/no-unresolved
  4:35  error  Unable to resolve path to module './helpers/withSetPropAction'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/navigators/AppNavigator.tsx
  11:27  error  Unable to resolve path to module '../models'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/DemoDebugScreen.tsx
  17:27  error  Unable to resolve path to module '../models'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/DemoPodcastListScreen.tsx
   1:26  error  Unable to resolve path to module 'mobx-react-lite'  import/no-unresolved
  35:27  error  Unable to resolve path to module '../models'        import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/LoginScreen.tsx
   1:26  error  Unable to resolve path to module 'mobx-react-lite'  import/no-unresolved
  12:27  error  Unable to resolve path to module '../models'        import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/WelcomeScreen.tsx
  5:27  error  Unable to resolve path to module '../models'  import/no-unresolved

✖ 12 problems (12 errors, 0 warnings)
```

These errors show the specific files and lines where MobX code or related imports were not fully removed by the tool.

# Options when Igniting project:

```
✅ What bundle identifier? · com.projectname
✅ Where do you want to start your project? · /Users/username/workspace/project-folder/project-name
✅ How do you want to manage Native code? · cng
✅ Do you want to initialize a git repository? (Y/n) · Yes
✅ Which package manager do you want to use? · bun
✅ Do you want to install dependencies? (Y/n) · Yes
✅ [Experimental] Expo Router for navigation? (This will remove the demo application) (y/N) · No
✅ Enable the New Architecture? (y/N) · Yes
✅ Include MobX-State-Tree for state management? (Recommended - opting out will remove the demo application) (Y/n) · No
   Warning: the demo application will be removed.

   
    █ Creating project-name using Ignite 10.4.0
    █ Powered by  ∞ Infinite Red  (https://infinite.red/)
    █ Package Manager: bun
    █ Bundle identifier: com.projectname
    █ Path: /Users/username/workspace/project-folder/project-name
    ────────────────────────────────────────────────────
   
    🖨  3D-printing a new React Native app
    🎨 Getting those last few details perfect
    🧶 Installing bun dependencies (wow these are heavy)
    ⚙️ Configuring app.json
    🛠️ Generating native template via Expo Prebuild
    🛠️ Removing fancy demo markup
    🌳 Removing MobX-State-Tree code
    ❌ Cleaning up
    
The following error occurred:
    
    $ eslint . --fix

/Users/username/workspace/project-folder/project-name/app/models/AuthenticationStore.ts
  1:46  error  Unable to resolve path to module 'mobx-state-tree'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/models/Episode.ts
  1:58  error  Unable to resolve path to module 'mobx-state-tree'              import/no-unresolved
  2:35  error  Unable to resolve path to module './helpers/withSetPropAction'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/models/EpisodeStore.ts
  1:46  error  Unable to resolve path to module 'mobx-state-tree'              import/no-unresolved
  4:35  error  Unable to resolve path to module './helpers/withSetPropAction'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/navigators/AppNavigator.tsx
  11:27  error  Unable to resolve path to module '../models'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/DemoDebugScreen.tsx
  17:27  error  Unable to resolve path to module '../models'  import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/DemoPodcastListScreen.tsx
   1:26  error  Unable to resolve path to module 'mobx-react-lite'  import/no-unresolved
  35:27  error  Unable to resolve path to module '../models'        import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/LoginScreen.tsx
   1:26  error  Unable to resolve path to module 'mobx-react-lite'  import/no-unresolved
  12:27  error  Unable to resolve path to module '../models'        import/no-unresolved

/Users/username/workspace/project-folder/project-name/app/screens/WelcomeScreen.tsx
  5:27  error  Unable to resolve path to module '../models'  import/no-unresolved

✖ 12 problems (12 errors, 0 warnings)

error: script "lint" exited with code 1

    
    Consider opening an issue with the following information at:
    https://github.com/infinitered/ignite/issues/new?template=bug_report.yml&labels=bug
    
    ⠦  Gathering system and project details   Unable to gather system and project details.
```

### Ignite version

10.4.0

### Additional info

Fails when I run it:

```
> npx ignite-cli doctor
Need to install the following packages:
ignite-cli@10.4.0
Ok to proceed? (y) y

System
  platform           darwin                                                 
  arch               arm64                                                  
  cpu                14 cores     Apple M3 Max                              
  directory          project-dir  /Users/anon/workspace/project-dir 
/Users/anon/.npm/_npx/e31027f3785124a8/node_modules/gluegun/build/index.js:15
    throw up;
    ^

Error: Command failed: pnpm --version
```

--- a/src/commands/new.ts
+++ b/src/commands/new.ts
@@ -514,6 +514,7 @@ module.exports = {
     let removeDemo = useDefault(options.removeDemo)
       ? defaultRemoveDemo
       : boolFlag(options.removeDemo)
+
     if (!defaultRemoveDemo && removeDemo === undefined) {
       const removeDemoResponse = await prompt.ask<{ removeDemo: boolean }>(() => ({
         type: "confirm",
@@ -525,6 +526,8 @@ module.exports = {
         prefix,
       }))
       removeDemo = removeDemoResponse.removeDemo
+    } else {
+      removeDemo = defaultRemoveDemo
     }
     // #endregion
 
@@ -826,6 +829,10 @@ module.exports = {
       // #region Remove Demo code
       const removeDemoPart = removeDemo === true ? "code" : "markup"
       startSpinner(`Removing fancy demo ${removeDemoPart}`)
+
+      p(yellow(`removeDemo: ${removeDemo}`))
+      p(yellow(`removeDemoPart: ${removeDemoPart}`))
+      // process.exit(1)
       try {
         const IGNITE = "node " + filesystem.path(__dirname, "..", "..", "bin", "ignite")
         const CMD = removeDemo === true ? "remove-demo" : "remove-demo-markup"

